mteb/tasks/Retrieval/multilingual/MLQARetrieval.py

```diff
@@ -0,0 +1,152 @@
+    "mlqa.vi.es": ["vie-Latn", "spa-Latn"],
+    "mlqa.vi.hi": ["vie-Latn", "hin-Deva"],
+    "mlqa.vi.vi": ["vie-Latn", "vie-Latn"],
+    "mlqa.vi.zh": ["vie-Latn", "zho-Hans"],
+    "mlqa.zh.ar": ["zho-Hans", "ara-Arab"],
+    "mlqa.zh.de": ["zho-Hans", "deu-Latn"],
+    "mlqa.zh.en": ["zho-Hans", "eng-Latn"],
+    "mlqa.zh.es": ["zho-Hans", "spa-Latn"],
+    "mlqa.zh.hi": ["zho-Hans", "hin-Deva"],
+    "mlqa.zh.vi": ["zho-Hans", "vie-Latn"],
+    "mlqa.zh.zh": ["zho-Hans", "zho-Hans"],
+}
+
+
+def extend_lang_pairs() -> dict[str, list[str]]:
+    eval_langs = {}
+    for langs in _LANGUAGES.values():
+        lang_pair = langs[0] + "_" + langs[1]
+        eval_langs[lang_pair] = langs
+    return eval_langs
+
+
+_EVAL_LANGS = extend_lang_pairs()
+
+
+class MLQARetrieval(AbsTaskRetrieval, CrosslingualTask):
+    metadata = TaskMetadata(
+        name="MLQARetrieval",
+        description="""MLQA (MultiLingual Question Answering) is a benchmark dataset for evaluating cross-lingual question answering performance.
+        MLQA consists of over 5K extractive QA instances (12K in English) in SQuAD format in seven languages - English, Arabic,
+        German, Spanish, Hindi, Vietnamese and Simplified Chinese. MLQA is highly parallel, with QA instances parallel between
+        4 different languages on average.""",
+        reference="https://huggingface.co/datasets/mlqa",
+        dataset={
+            "path": "mlqa",
+            "revision": "397ed406c1a7902140303e7faf60fff35b58d285",
+        },
+        type="Retrieval",
+        category="s2p",
+        eval_splits=["validation", "test"],
+        eval_langs=_EVAL_LANGS,
+        main_score="ndcg_at_10",
+        date=("2019-01-01", "2020-12-31"),
+        form=["written"],
+        domains=["Encyclopaedic"],
+        task_subtypes=["Question answering"],
+        license="cc-by-sa-3.0",
+        socioeconomic_status="mixed",
+        annotations_creators="human-annotated",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="""@article{lewis2019mlqa,
+        title = {MLQA: Evaluating Cross-lingual Extractive Question Answering},
+        author = {Lewis, Patrick and Oguz, Barlas and Rinott, Ruty and Riedel, Sebastian and Schwenk, Holger},
+        journal = {arXiv preprint arXiv:1910.07475},
+        year = 2019,
+        eid = {arXiv: 1910.07475}
+        }""",
+        n_samples={"test": 158083, "validation": 15747},
+        avg_character_length={
+            "test": 37352.28,
+            "validation": 36952.7,
+        },  # avergae context lengths
+    )
+
+    def load_data(self, **kwargs):
+        if self.data_loaded:
+            return
+
+        _dataset_raw = {}
+        self.queries, self.corpus, self.relevant_docs = {}, {}, {}
+
+        for hf_subset, langs in _LANGUAGES.items():
+            lang_pair = (
+                langs[0] + "_" + langs[1]
+            )  # builds a language pair separated by an underscore. e.g., "ara-Arab_eng-Latn"
```

Comment:
What do you think about, instead of `eng-Latn_fra-Latn`, we only use the 3 letter lang code, i.e. `eng_fra` ? As they should be unique, it saves a lot of typing. 

Aside: could also consider making line 118 into its own small function as it's used more than once.

Reply:
I like this proposition, I'm afraid the format is not adequate to our language standard. I'll give it a try and let you know!

Reply:
It should be fine. When passed into `-l`, MTEB calls `_extend_lang_code`, which does a mapping if only a 3 letter code is provided. So both standards should be supported without problems.